File: backend/app/schemas/installment.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TripInstallmentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: str
    trip_id: str
    member_id: str
    month_index: int
    amount_due: float
    amount_paid: float
    paid_at: Optional[datetime] = None
    created_at: datetime


class TripInstallmentPay(BaseModel):
    amount_paid: float


class TripInstallmentAdjust(BaseModel):
    new_months: int  # jumlah bulan baru

File: frontend/src/types/installment.ts
export interface TripInstallment {
  id: string;
  trip_id: string;
  member_id: string;
  month_index: number;
  amount_due: number;
  amount_paid: number;
  paid_at: string | null;
  created_at: string;
}

export interface TripInstallmentPayRequest {
  // Jumlah yang dibayarkan kali ini — backend MENAMBAHKAN nilai ini ke
  // amount_paid yang sudah ada, bukan menggantinya.
  amount_paid: number;
}

export interface TripInstallmentAdjustRequest {
  new_months: number;
}
